internal/modules/tesla/tesla.go

```diff
@@ -43,7 +44,16 @@ func (m Module) CloudEventConvert(_ context.Context, msgData []byte) ([]cloudeve
 		return nil, nil, fmt.Errorf("failed to unmarshal message: %w", err)
 	}
 	hdrs := []cloudevent.CloudEventHeader{event.CloudEventHeader}
-	if gjson.GetBytes(event.Data, "vin").Exists() {
+	if event.DataVersion == tesla.FleetTelemetryDataVersion {
+		payloads := gjson.GetBytes(event.Data, "payloads")
+		// This check should always pass.
+		if payloads.Exists() && payloads.IsArray() && len(payloads.Array()) != 0 {
+			fpHdr := event.CloudEventHeader
+			fpHdr.Type = cloudevent.TypeFingerprint
+			hdrs = append(hdrs, fpHdr)
+		}
+	} else if gjson.GetBytes(event.Data, "vin").Exists() {
+		// Must be a Fleet API response.
```

Comment:
Don't feel great about the duplication or the gjson construction, but it seems to work.